connector/connect/common/src/main/protobuf/spark/connect/commands.proto

```diff
@@ -321,6 +324,50 @@ message StreamingQueryCommandResult {
   }
 }
 
+// Commands for the streaming query manager.
+message StreamingQueryManagerCommand {
+
+  // See documentation for the corresponding API method in StreamingQueryManager.
+  oneof command {
+    // active() API, returns a list of active queries.
+    bool active = 1;
+    // get() API, returns the StreamingQuery identified by id.
+    GetQueryCommand get_query = 2;
+    // awaitAnyTermination() API, wait until any query terminates or timeout.
+    AwaitAnyTerminationCommand await_any_termination = 3;
+    // resetTerminated() API.
+    bool reset_terminated = 4;
+  }
+
+  message GetQueryCommand {
+    // (Required) The unique id of the query that it wants to get.
+    string id = 1;
+  }
+
+  message AwaitAnyTerminationCommand {
+    // (Optional) The waiting time in milliseconds to wait for any query to terminate.
+    optional int64 timeout_ms = 1;
+  }
+}
+
+// Response for commands on the streaming query manager.
+message StreamingQueryManagerCommandResult {
+  oneof result_type {
+    ActiveResult active = 1;
+    StreamingQueryInstanceId query = 2;
+    AwaitAnyTerminationResult await_any_termination = 3;
+    bool reset_terminated = 4;
+  }
+
+  message ActiveResult {
+    repeated StreamingQueryInstanceId active_queries = 1;
```

Comment:
This can be the oneof field.

Reply:
Yes that's my original thought. But I got IntelliJ warning `Oneof fields cannot have labels` so that's why I ended up doing it like this

Reply:
Oh, that might be. Does it fail to compile. We can keep as you have now.